### ISSUE TITLE
1.2.2 Door prompt giving error for a user in Dallas

### ISSUE THREAD

**Issue by kategodfrey** [CCDevNelson/Snap-Updates#293]
Hi @ClassyTeddy or @SeanRadvan  or anyone who can help 😂

I received a call from a user saying he is not able to put doors in. He did the update and followed our instructions on fixing the leftover files and now he gets an error any time he tries to put in doors.  Here is his error:

![Screenshot 2021-01-22 174834](https://user-images.githubusercontent.com/63115502/105561945-855a2900-5cd5-11eb-815a-e2da076a21bf.jpg)

**Comment by RyanMDev** [comment]
It looks to me like he may have some of the old pull thumbnails/.png images leftover from version 1.2.1.

The "Matte Nickel" pull category has been removed in version 1.2.2. So it appears that he still has the old thumbnail image but the .blend file for the pull no longer exists. This could be why it is looking for the old pull.

Can you confirm this?

The pulls folder is located here:
`SNaP\2.78\scripts\addons\snap_db\library_scripts\Library-Classy_Closets\Door Pulls`

The following folders should **not** be inside:
`Black`, `Bronze`, `Chrome`, `Matte Chrome`, `Matte Nickel`, `Nickel`, `Stainless`

Additionally, every .png image for a pull should have a matching .blend file.

I suspect that there are some old thumbnails/.png images hanging around that do not have matching .blend files because they have been removed.

**Comment by RyanMDev** [comment]
Alright, I'll make a fix for this and include it in the hotfix update with the 2d views fix.

**Comment by kategodfrey** [comment]
Great. Thank you both!

**Comment by kategodfrey** [comment]
I just got this error after selecting left swing. Happened 2 times in a row.

![image](https://user-images.githubusercontent.com/63115502/106042126-de410d00-6099-11eb-8949-f3041afcbfc2.png)

**Comment by kategodfrey** [comment]
Got this one after selecting "fill opening." I believe Ted opened an issue about the fill opening issue... I can search and see which issue # it is.
 
![image](https://user-images.githubusercontent.com/63115502/106042235-003a8f80-609a-11eb-8a9b-d4c75dfec3bd.png)

**Comment by ClassyTeddy** [comment]
The issue actually has to do with the doors looking for a parent object that doesn't exist. It looks like the logic isn't working because you put the door in an island.

**Comment by RyanMDev** [comment]
I am able to reproduce this on the master branch.

Both of the errors above come from the same line of code in `data_closet_doors.py` (line: 618). 

        if(self.assembly.obj_bp.parent.lm_closets.is_closet):
            closet_obj = self.assembly.obj_bp.parent
            closet_assembly = fd_types.Assembly(closet_obj)
        elif(self.assembly.obj_bp.parent.parent.lm_closets.is_closet):
            closet_obj = self.assembly.obj_bp.parent.parent
            closet_assembly = fd_types.Assembly(closet_obj)
        elif(self.assembly.obj_bp.parent.parent.parent):
            if(self.assembly.obj_bp.parent.parent.parent.lm_closets.is_closet):
                closet_obj = self.assembly.obj_bp.parent.parent.parent
                closet_assembly = fd_types.Assembly(closet_obj)

This block of code needs to use the `get_parent_assembly_bp` function to get the parent assembly bp instead of relying on the above code to check for a parent that may or may not exist.

@ClassyTeddy, use this instead:
```
        closet_obj = utils.get_parent_assembly_bp(self.assembly.obj_bp)
        if closet_obj.lm_closets.is_closet:
            closet_assembly = fd_types.Assembly(closet_obj)
```

Also see commit: CCDevNelson/SNaP-Dev@cf51de3c343182facbc8f50a87a7cf15f012653b. Here I fixed this in two other areas in this file.

Let's create a separate issue for this.

This issue here involves an update not removing old files. The two error messages above exist on the master branch and are related to the doors prompt menu in `data_closet_doors.py`.

**Comment by RyanMDev** [comment]
I have included the above fix on commit: CCDevNelson/SNaP-Dev@42df5485805c9512852ea0c7e56e860134ae2c1f
